db/db_compaction_test.cc

```diff
@@ -7958,6 +7958,42 @@ TEST_F(DBCompactionTest, BottomPriCompactionCountsTowardConcurrencyLimit) {
   compact_range_thread.join();
 }
 
+TEST_F(DBCompactionTest, BottommostFileCompactionAllowIngestBehind) {
+  // allow_ingest_behind prevents seqnum zeroing, and could cause
+  // compaction loop with reason kBottommostFiles.
+  Options options = CurrentOptions();
+  options.env = env_;
+  options.compaction_style = kCompactionStyleLevel;
+  options.allow_ingest_behind = true;
+  options.comparator = BytewiseComparator();
+  DestroyAndReopen(options);
+
+  WriteOptions write_opts;
+  ASSERT_OK(db_->Put(write_opts, "infinite", "compaction loop"));
+  ASSERT_OK(db_->Put(write_opts, "infinite", "loop"));
+
+  ASSERT_OK(Flush());
+  MoveFilesToLevel(1);
+  ASSERT_OK(db_->Put(write_opts, "bumpseqnum", ""));
+  ASSERT_OK(Flush());
+  auto snapshot = db_->GetSnapshot();
+  // Bump up oldest_snapshot_seqnum_ in VersionStorageInfo.
+  db_->ReleaseSnapshot(snapshot);
```

Comment:
Does `ComputeBottommostFilesMarkedForCompaction()` pick the L1 file even without snapshot?

Reply:
I think so. The criteria in `ComputeBottommostFilesMarkedForCompaction` is satisfied. The sequence number in the L1 file is not zeroed out due to allow_ingest_behind.

Reply:
If you mean not taking snapshot in this test, then no. Since oldest_snapshot_seqnum_ would be 0 and only files with 0 < largest_seqno < oldest_snapshot_seqnum_ is compacted.

Reply:
Maybe `oldest_snapshot_seqnum_` isn't set properly when there's no snapshots. Anyways it's not very important.